go/teams/chain_test.go

```diff
@@ -59,6 +60,87 @@ func TestTeamSigChainParse(t *testing.T) {
+	tc := SetupTest(t, "team_sig_chain_high_links", 1)
+	defer tc.Cleanup()
+	ctx := context.TODO()
+
+	// Create some users. The owner is last so that it has the active session.
+	u2, err := kbtest.CreateAndSignupFakeUser("we", tc.G) //admin
+	require.NoError(t, err)
+	u3, err := kbtest.CreateAndSignupFakeUser("ji", tc.G) //non-admin
+	require.NoError(t, err)
+	u1, err := kbtest.CreateAndSignupFakeUser("je", tc.G) //owner
+	require.NoError(t, err)
+
+	// Create a team. This creates the first high link.
+	teamName := u1.Username + "t"
+	teamNameObj, err := keybase1.TeamNameFromString(teamName)
+	require.NoError(t, err)
+	teamID, err := CreateRootTeam(ctx, tc.G, teamName, keybase1.TeamSettings{})
+	require.NoError(t, err)
+	assertHighSeqForTeam(t, tc, teamID, 1)
+
+	// Adding a new reader is not a high link, so the lastest high seq won't change.
+	_, err = AddMember(ctx, tc.G, teamName, u3.Username, keybase1.TeamRole_READER)
+	require.NoError(t, err)
+	assertHighSeqForTeam(t, tc, teamID, 1)
+
+	// Adding a new admin IS a high link, so we should jump to 3.
+	_, err = AddMember(ctx, tc.G, teamName, u2.Username, keybase1.TeamRole_ADMIN)
+	require.NoError(t, err)
+	assertHighSeqForTeam(t, tc, teamID, 3)
+
+	// Creating a subteam is not a high link for the parent team
+	// but it is for the subteam. The link types are different
+	// "team.root" and "team.subteam_head" so we should check both teams.
+	sub := "sub"
+	subteamID, err := CreateSubteam(ctx, tc.G, sub, teamNameObj, keybase1.TeamRole_ADMIN)
+	require.NoError(t, err)
+	assertHighSeqForTeam(t, tc, subteamID, 1)
+	assertHighSeqForTeam(t, tc, teamID, 3)
+
+	// Adding an admin to the subteam is a high link for the subteam but not
+	// the parent. Primarily, we do this to verify that high links advance
+	// the same way on subteams (since there are places that default to a
+	// value of 1 for sequence number). It's overkill to test any more than
+	// just this on the subteam since it should work the same way.
+	_, err = AddMemberByID(ctx, tc.G, *subteamID, u3.Username, keybase1.TeamRole_ADMIN)
+	require.NoError(t, err)
+	assertHighSeqForTeam(t, tc, subteamID, 2)
+	assertHighSeqForTeam(t, tc, teamID, 3)
+
+	// Back to the root team... downgrading an admin IS a high link for the root team
+	// but it is not one for the subteam, because the subteam needs to care about it's
+	// parent's high links anyway.
+	err = EditMember(ctx, tc.G, teamName, u2.Username, keybase1.TeamRole_WRITER)
+	require.NoError(t, err)
+	assertHighSeqForTeam(t, tc, teamID, 5)
+	assertHighSeqForTeam(t, tc, subteamID, 2)
+
+	// Rotated keys do not create high links.
+	err = RotateKey(ctx, tc.G, *teamID)
+	require.NoError(t, err)
+	assertHighSeqForTeam(t, tc, teamID, 5)
+	assertHighSeqForTeam(t, tc, subteamID, 2)
+
+	// Deleting a subteam will not create a high link.
+	subteamName := teamName + "." + sub
+	err = Delete(ctx, tc.G, &teamsUI{}, subteamName)
+	require.NoError(t, err)
+	assertHighSeqForTeam(t, tc, teamID, 5)
+}
```

Comment:
maybe a test for readding an admin after the admin resets? It should work because you're comparing `UserVersion`s though